torch/_dynamo/symbolic_convert.py

```diff
@@ -949,6 +949,61 @@ def COMPARE_OP(self, inst):
                     **options,
                 )
             )
+        elif (
+            isinstance(left, BaseListVariable)
+            and all(
+                isinstance(l, (ConstantVariable, DynamicShapeVariable))
+                for l in left.items
+            )
+            and isinstance(right, BaseListVariable)
+            and all(
+                isinstance(r, (ConstantVariable, DynamicShapeVariable))
+                for r in right.items
+            )
+            and len(left.items) == len(right.items)
+            and op in supported_tensors
+        ):
+            # Invariant: at least one of the variables in the two lists being compared is a DynamicShape.
+            # If all were constants, then we would have hit the `left.is_python_constant() and right.is_python_constant()`
+            # case further up.
+            # This means that the output here will always be a DynamicShapeVariable.
+
+            # Special case lists, which are not represented directly as proxies.
+            # Instead, we need to take our list of proxies and manually reduce it.
+            if isinstance(left, ListVariable) and isinstance(right, ListVariable):
+                list_variable = [
+                    DynamicShapeVariable.create(
+                        self,
+                        supported_tensors[op](l.as_proxy(), r.as_proxy()),
+                        dyn_shape=None,
+                        **options,
+                    )
+                    for l, r in zip(left.items, right.items)
+                ]
+                # list_variable represents a pointwise comparison of our list elements.
+                # Now reduce across the list.
+                dyn_variable = functools.reduce(
+                    lambda a, b: DynamicShapeVariable.create(
+                        self,
+                        # TODO: morally, this should be "a and b",
+                        # but we don't have SymBool yet,
+                        # and we don't have "and" implemented on SymInts.
+                        # + should give us the same output though.
+                        operator.and_(a.as_proxy(), b.as_proxy()),
+                        dyn_shape=None,  # TODO: what's this do / do I need to mess with it
+                        **options,
+                    ),
+                    list_variable,
+                )
+            else:
+                dyn_variable = DynamicShapeVariable.create(
```

Comment:
Creating a DynamicShapeVariable here is kind of pointless, because the comparison ops MUST return a bool, which means you have guarded on the result of the comparison here. I think guarding is the right thing to do here, and you should just resolve this down to a constant at this point (with the understanding that the guards will help you out here later)

Reply:
So my thought was - once SymBool lands, then it'll be okay for the comparison op to return a SymBool, thus requiring our tracker to be a DynamicShapeVariable. Is that wrong? (comparison ops on SymInts should unconditionally materialize the concrete bool).

I guess in practice, almost every comparison on SymInts will do some kind of branching on the bool afterwards

Reply:
Yeah, I think in principle it is better to support symbolic bools in dynamo and propagate as far as they can.